backEnd/src/main/java/phylogenetictree/PhylogeneticNode.java

```diff
@@ -24,21 +30,41 @@ public PhylogeneticNode() {
         nameLabel = "";
         distance = 0.;
         children = new ArrayList<>();
+        genomes= new ArrayList<>();
     }
 
     /**
-     * Initialize this node from tree node.
+     * Initialize this node from tree node, recursively.
      * @param node The node.
+     * @param parent The parentNode of this node, null if root.
      * @param distance The distance.
+     * @param id The id of this node, root being 0. Incrementing breath first.
      */
-    public PhylogeneticNode(TreeNode node, double distance) {
+    public PhylogeneticNode(TreeNode node, PhylogeneticNode parent, double distance, int id) {
```

Comment:
The Parent is needed to iterate upwards through the tree in order to store the genomes at their specific treeNodes. The id is simply a way to identify nodes without a label.